File: 3001-3250/3120. Count the Number of Special Characters I/solution.java
class Solution {
    public int numberOfSpecialChars(String word) {
        int[] min = new int[26] ; 
        int[] maj = new int[26] ;
        for( char c : word.toCharArray() ){
            if( c>='a' && c<='z'){
                min[ c - 'a']++ ; 
            }else {
                maj[ c - 'A' ]++ ; 
            }
        } 
        int rs = 0 ; 
        for(int i=0 ; i<26 ; ++i){
            if( min[i]>0 && maj[i]>0 ) rs++ ; 
        }
        return rs ; 
    }
}

File: 3001-3250/3120. Count the Number of Special Characters I/solution.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
    int numberOfSpecialChars(string word)
    {
        int n = word.size();
        vector<bool> upper(26, false);
        vector<bool> lower(26, false);
        for (int i = 0; i < n; ++i)
        {
            if (word[i] > 'Z')
            {
                lower[word[i] - 'a'] = true;
            }
            else
            {
                upper[word[i] - 'A'] = true;
            }
        }

        int ans = 0;
        for (int i = 0; i < 26; ++i)
        {
            if (upper[i] && lower[i])
            {
                ans++;
            }
        }
        return ans;
    }
};